src/HDF5.jl

```diff
@@ -36,6 +36,12 @@ else
     error("HDF5 not properly installed. Please run Pkg.build(\"HDF5\")")
 end
 
+let d = dirname(libhdf5), b = basename(libhdf5)
+    base,ext = split(b, ".", limit=2)
+    hl = joinpath(d, "libhdf5_hl.$ext")
+    global const libhdf5_hl = isfile(hl) ? hl : ""
+end
```

Comment:
why not something simpler (setting the sentinel value to `nothing` instead of `""`)
```julia
hdf5_hl = joinpath(dirname(libhdf5), "libhdf5_hl.$(Libdl.dlext)")
const libhdf5_hl = isfile(hdf5_hl ) ? hdf5_hl : nothing
```